src/site/downloads/archive/web/download_archive.dart

```diff
@@ -2,103 +2,160 @@ import 'dart:async';
+  'stable': querySelector('#stable-os'),
+  'dev': querySelector('#dev-os')
 };
 
 void main() {
-  HttpRequest.getString("$storageApiBase?prefix=channels/stable/release/&delimiter=/")
-      .then((resp) { getListing('stable', resp); });
-  HttpRequest.getString("$storageApiBase?prefix=channels/dev/release/&delimiter=/")
-      .then((resp) { getListing('dev', resp); });
-  
-  versionSelectors['stable'].onChange.listen((Event event) { filterTable('stable', event); });
-  versionSelectors['dev'].onChange.listen((Event event) { filterTable('dev', event); });
-  osSelectors['stable'].onChange.listen((Event event) { filterTable('stable', event); });
-  osSelectors['dev'].onChange.listen((Event event) { filterTable('dev', event); });
+  HttpRequest
+      .getString("$storageApiBase?prefix=channels/stable/release/&delimiter=/")
+      .then((resp) {
+    getListing('stable', resp);
+  });
+  HttpRequest
+      .getString("$storageApiBase?prefix=channels/dev/release/&delimiter=/")
+      .then((resp) {
+    getListing('dev', resp);
+  });
+
+  versionSelectors['stable'].onChange.listen((Event event) {
+    filterTable('stable', event);
+  });
+  versionSelectors['dev'].onChange.listen((Event event) {
+    filterTable('dev', event);
+  });
+  osSelectors['stable'].onChange.listen((Event event) {
+    filterTable('stable', event);
+  });
+  osSelectors['dev'].onChange.listen((Event event) {
+    filterTable('dev', event);
+  });
 }
 
 void filterTable(String channel, Event event) {
-  String selectedVersion = versionSelectors[channel].selectedOptions[0].attributes['value'];
-  String selectedOs = osSelectors[channel].selectedOptions[0].attributes['value'];
+  String selectedVersion =
+      versionSelectors[channel].selectedOptions[0].attributes['value'];
+  String selectedOs =
+      osSelectors[channel].selectedOptions[0].attributes['value'];
   if (selectedVersion == 'all' && selectedOs == 'all') {
-    tables[channel].querySelectorAll('tr[data-version]').classes.remove('hidden');
+    tables[channel].querySelectorAll('tr[data-version]').classes
+        .remove('hidden');
   } else {
     tables[channel].querySelectorAll('tr[data-version]').classes.add('hidden');
     String selector = 'tr';
-    if (selectedVersion != 'all') { selector += '[data-version="$selectedVersion"]'; }
-    tables[channel].querySelectorAll(selector+'[data-os="api"]').classes.remove('hidden');
-    if (selectedOs != 'all') { selector += '[data-os="$selectedOs"]'; }
+    if (selectedVersion != 'all') {
+      selector += '[data-version="$selectedVersion"]';
+    }
+    tables[channel].querySelectorAll(selector + '[data-os="api"]').classes
+        .remove('hidden');
+    if (selectedOs != 'all') {
+      selector += '[data-os="$selectedOs"]';
+    }
     tables[channel].querySelectorAll(selector).classes.remove('hidden');
   }
 }
 
+DateTime parseDateTime(String date) {
+  try {
+    return DateTime.parse(date);
+  } catch (_) {}
+
+  if (date.length == 12) {
```

Comment:
is there a valid format that will have 12 chars? if not, we could do this check first, and otherwise return the parsed date (and leave the throw uncaught at that point)